use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/rest/VehicleRouteDemoResource.java

```diff
@@ -0,0 +1,225 @@
+            if (customerCount < 1) {
+                throw new IllegalStateException(
+                        "Number of customerCount (" + customerCount + ") must be greater than zero.");
+            }
+            if (vehicleCount < 1) {
+                throw new IllegalStateException(
+                        "Number of vehicleCount (" + vehicleCount + ") must be greater than zero.");
+            }
+            if (depotCount < 1) {
+                throw new IllegalStateException(
+                        "Number of depotCount (" + depotCount + ") must be greater than zero.");
+            }
+            if (northEastCorner.getLatitude() <= southWestCorner.getLatitude()) {
+                throw new IllegalStateException("northEastCorner.getLatitude (" + northEastCorner.getLatitude()
+                        + ") must be greater than southWestCorner.getLatitude(" + southWestCorner.getLatitude() + ").");
+            }
+            if (northEastCorner.getLongitude() <= southWestCorner.getLongitude()) {
+                throw new IllegalStateException("northEastCorner.getLongitude (" + northEastCorner.getLongitude()
+                        + ") must be greater than southWestCorner.getLongitude(" + southWestCorner.getLongitude() + ").");
+            }
+
+            this.seed = seed;
+            this.customerCount = customerCount;
+            this.vehicleCount = vehicleCount;
+            this.depotCount = depotCount;
+            this.vehicleStartTime = vehicleStartTime;
+            this.minDemand = minDemand;
+            this.maxDemand = maxDemand;
+            this.minVehicleCapacity = minVehicleCapacity;
+            this.maxVehicleCapacity = maxVehicleCapacity;
+            this.southWestCorner = southWestCorner;
+            this.northEastCorner = northEastCorner;
+        }
+    }
+
+    @APIResponses(value = {
+            @APIResponse(responseCode = "200", description = "List of demo data represented as IDs.",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON,
+                            schema = @Schema(implementation = DemoData.class, type = SchemaType.ARRAY))) })
+    @Operation(summary = "List demo data.")
+    @GET
+    public DemoData[] list() {
+        return DemoData.values();
+    }
+
+    @APIResponses(value = {
+            @APIResponse(responseCode = "200", description = "Unsolved demo route plan.",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON,
+                            schema = @Schema(implementation = VehicleRoutePlan.class))) })
+    @Operation(summary = "Find an unsolved demo route plan by ID.")
+    @GET
+    @Path("/{demoDataId}")
+    public VehicleRoutePlan generate(@Parameter(description = "Unique identifier of the demo data.",
+            required = true) @PathParam("demoDataId") DemoData demoData) {
+        return build(demoData);
+    }
+
+    public VehicleRoutePlan build(DemoData demoData) {
+        String name = "demo";
```

Comment:
I'm thinking... `var`? The methoud could arguably use it, but maybe for consistency with the rest of the code, we don't?

Reply:
I would probably leave it as it is because of consistency, but up to you.